addons/mail/static/src/components/attachment_delete_confirm_dialog/attachment_delete_confirm_dialog.js

```diff
@@ -73,6 +73,7 @@ class AttachmentDeleteConfirmDialog extends Component {
         this._dialogRef.comp._close();
         this.attachment.remove();
         this.trigger('o-attachment-removed', { attachmentLocalId: this.props.attachmentLocalId });
+        this.trigger('reload', { keepChanges: true });
```

Comment:
This should trigger `o_attachments_changed` instead so that it properly triggers reload only if `hasRecordReloadOnAttachmentsChanged` is `true`.